plugins/aws/autoscaling/sameAzElb.js

```diff
@@ -0,0 +1,117 @@
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+        var loadBalancers = {};
+
+        async.each(regions.autoscaling, function(region, rcb){
+            var autoScalingGroups = helpers.addSource(cache, source,
+                ['autoscaling', 'describeAutoScalingGroups', region]);
+                
+            var elasticLoadBalancers = helpers.addSource(cache, source,
+                ['elb', 'describeLoadBalancers', region]);
+
+            var elasticLoadBalancersV2 = helpers.addSource(cache, source,
+                ['elbv2', 'describeLoadBalancers', region]);
+
+            if (!autoScalingGroups || !elasticLoadBalancers || !elasticLoadBalancersV2) return rcb();
+
+            if (autoScalingGroups.err || !autoScalingGroups.data) {
+                helpers.addResult(results, 3, 'Unable to query for AutoScaling groups: ' +  helpers.addError(autoScalingGroups), region);
+                return rcb();
+            }
+
+            if (elasticLoadBalancers.err || !elasticLoadBalancers.data) {
+                helpers.addResult(results, 3, 'Unable to query for Classic load balancers: ' +  helpers.addError(elasticLoadBalancers), region);
+                return rcb();
+            }
+
+            if (elasticLoadBalancersV2.err || !elasticLoadBalancersV2.data) {
+                helpers.addResult(results, 3, 'Unable to query for Application/Network load balancers: ' +  helpers.addError(elasticLoadBalancersV2), region);
+                return rcb();
+            }
+
+            if (!autoScalingGroups.data.length) {
+                helpers.addResult(results, 0, 'No AutoScaling group found', region);
+                return rcb();
+            }
+            
+            if (elasticLoadBalancers.data.length) {
+                elasticLoadBalancers.data.forEach(function(elb) {
+                    if(elb.LoadBalancerName) {
+                        loadBalancers[elb.LoadBalancerName] =  elb;
+                    }
+                });
+            }
+
+            if (elasticLoadBalancersV2.data.length) {
+                elasticLoadBalancersV2.data.forEach(function(elbv2) {
+                    if(elbv2.LoadBalancerName) {
+                        loadBalancers[elbv2.LoadBalancerName] =  elbv2;
+                    }
+                });
+            }
+
+            autoScalingGroups.data.forEach(function(asg) {
+                var asgAvailabilityZones = asg.AvailabilityZones;
+                var distinctAzs = [];
+                var resource = asg.AutoScalingGroupARN;
+
+                if(asg.HealthCheckType == 'ELB') {
+                    if (asg.LoadBalancerNames && asg.LoadBalancerNames.length) {
+
+                        asg.LoadBalancerNames.forEach(function(elbName) {
+                            if(elbName in loadBalancers) {
```

Comment:
Can you replace this check with `if(loadBalancers[elbName]) {` this technically does the same thing but is more aligned with the codebase